### PR TITLE
add tencentos fact distro

### PR DESCRIPTION
**Pull request by jameslivulpi**
##### SUMMARY
takes care of #76343 

Correct the distribution fact if node is running TencentOS

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
`distribution` fact gethering

##### ADDITIONAL INFORMATION

Before this fix:

```
10.1.1.2 | SUCCESS => {
    "ansible_facts": {
        "ansible_distribution": "CentOS", <------
        "ansible_distribution_file_parsed": true,
        "ansible_distribution_file_path": "/etc/redhat-release",
        "ansible_distribution_file_variety": "RedHat",
        "ansible_distribution_major_version": "3",
        "ansible_distribution_release": "Final",
        "ansible_distribution_version": "3.1",
        "discovered_interpreter_python": "/usr/bin/python3.6"
    },
    "changed": false
}

```

test.yml:
```
- name: install socat & conntrack
    package:
      name:
        - socat
        - conntrack
      state: present
    become: true
```

Failed with:
```
fatal: [10.1.1.2]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "allow_downgrade": false,
            "autoremove": false,
            "bugfix": false,
            "cacheonly": false,
            "conf_file": null,
            "disable_excludes": null,
            "disable_gpg_check": false,
            "disable_plugin": [],
            "disablerepo": [],
            "download_dir": null,
            "download_only": false,
            "enable_plugin": [],
            "enablerepo": [],
            "exclude": [],
            "install_repoquery": true,
            "install_weak_deps": true,
            "installroot": "/",
            "list": null,
            "lock_timeout": 30,
            "name": [
                "socat",
                "conntrack"
            ],
            "releasever": null,
            "security": false,
            "skip_broken": false,
            "state": "present",
            "update_cache": false,
            "update_only": false,
            "use_backend": "auto",
            "validate_certs": true
        }
    },
    "msg": "The Python 2 yum module is needed for this module. If you require Python 3 support use the `dnf` Ansible module instead."
}
```

After fix:

```
10.1.1.2 | SUCCESS => {
    "ansible_facts": {
        "ansible_distribution": "TencentOS",
        "ansible_distribution_file_parsed": true,
        "ansible_distribution_file_path": "/etc/centos-release",
        "ansible_distribution_file_variety": "CentOS",
        "ansible_distribution_major_version": "3",
        "ansible_distribution_release": "Final",
        "ansible_distribution_version": "3.1",
        "discovered_interpreter_python": "/usr/bin/python3.6"
    },
    "changed": false
}
```

installs via dnf and succeeds:
```
changed: [10.1.1.2] => {
    "changed": true,
    "invocation": {
        "module_args": {
            "allow_downgrade": false,
            "allowerasing": false,
            "autoremove": false,
            "bugfix": false,
            "cacheonly": false,
            "conf_file": null,
            "disable_excludes": null,
            "disable_gpg_check": false,
            "disable_plugin": [],
            "disablerepo": [],
            "download_dir": null,
            "download_only": false,
            "enable_plugin": [],
            "enablerepo": [],
            "exclude": [],
            "install_repoquery": true,
            "install_weak_deps": true,
            "installroot": "/",
            "list": null,
            "lock_timeout": 30,
            "name": [
                "socat",
                "conntrack"
            ],
            "nobest": false,
            "releasever": null,
            "security": false,
            "skip_broken": false,
            "state": "present",
            "update_cache": false,
            "update_only": false,
            "validate_certs": true
        }
    },
    "msg": "",
    "rc": 0,
    "results": [
        "Installed: libnetfilter_conntrack-1.0.6-5.tl3.x86_64",
        "Installed: libnetfilter_cthelper-1.0.0-15.tl3.x86_64",
        "Installed: conntrack-tools-1.4.4-10.tl3.x86_64",
        "Installed: libnetfilter_cttimeout-1.0.0-11.tl3.x86_64",
        "Installed: libnetfilter_queue-1.0.4-3.el8.x86_64",
        "Installed: libnfnetlink-1.0.1-13.tl3.x86_64",
        "Installed: socat-1.7.4.1-1.el8.x86_64"
    ]
}
```

### CODE REVIEW

File: changelogs/fragments/76343-tencentos-dnf.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+- distribution - detect tencentos and gather correct fact on distro
```
Comment (Akasurde) [inline changelogs/fragments/76343-tencentos-dnf.yml]:
```suggestion
- distribution - detect tencentos and gather correct facts on the distro.
```

File: lib/ansible/module_utils/facts/system/distribution.py [hunk 2]
```diff
@@ -464,6 +464,10 @@ def parse_distribution_file_CentOS(self, name, data, path, collected_facts):
             centos_facts['distribution_release'] = 'Stream'
             return True, centos_facts
 
+        if "TencentOS Server" in data:
```
Comment (Akasurde) [inline lib/ansible/module_utils/facts/system/distribution.py:467]: Thanks for the PR. Could you please add a test case for this change by using https://github.com/ansible/ansible/blob/devel/hacking/tests/gen_distribution_version_testcase.py (Like https://github.com/ansible/ansible/tree/devel/test/units/module_utils/facts/system/distribution/fixtures)?

**Review by Akasurde — CHANGES_REQUESTED** [review]

File: lib/ansible/module_utils/facts/system/distribution.py (hunk 2 above)
Comment (jameslivulpi) [reply lib/ansible/module_utils/facts/system/distribution.py:467]: Yes I will take care of this tomorrow

Comment (jameslivulpi) [reply lib/ansible/module_utils/facts/system/distribution.py:467]: ok done.

**Review by Akasurde — COMMENTED** [review]
LGTM, waiting for others to comment.

**Review by mkrizek — APPROVED** [review]

**Pull request merged**